### ISSUE TITLE
Dockerfile for 8.5.0 has no named user

### ISSUE THREAD

**Issue by slhck** [elastic/apm-server#9495]
**APM Server version** (`apm-server version`): 8.5.0

**Description of the problem including expected versus actual behavior**:

The Docker image has no user name.

Previous versions used to have a named user called `apm-server`.

**Steps to reproduce**:

```
$ docker run --rm -it docker.elastic.co/apm/apm-server:8.5.0 whoami   
whoami: cannot find name for user ID 1000
```

Previously this worked:

```
$ docker run --rm -it docker.elastic.co/apm/apm-server:8.4.3 whoami
apm-server
```

Down the line this leads to errors using APM Server in a Docker Compose project, because I cannot start a container that I build with the latest Dockerfile:

```
Error response from daemon: unable to find user apm-server: no matching entries in passwd file
```

**Comment by axw** [comment]
@slhck we've made a bunch of simplifications to the Dockerfile, one of which was to remove the apm-server user addition. We're now specifying `USER 1000` in the Dockerfile.

Is the named user specifically needed for your use case? Can you share your docker-compose?

**Comment by slhck** [comment]
No problem. I can work around the missing username, I just feel like this might cause issues down the road for other folks in different situations.

> avoiding having to install OS packages just to add a user

Ah, I see. I thought this was always included? So you mean something like for alpine, where you need `adduser` vs `useradd` for Debian-based images? I guess you could do an `RUN if …` statement with the proper command depending on the base OS.

**Comment by axw** [comment]
> Ah, I see. I thought this was always included? So you mean something like for alpine, where you need adduser vs useradd for Debian-based images?

adduser is in Ubuntu (our default base image), but not in Red Hat UBI 8 -- and neither is useradd. We were installing it with microdnf, but this makes the build a bit less reliable and reproducible.

> I guess you could do an RUN if … statement with the proper command depending on the base OS.

Maybe. I'd like to avoid installing things for the above reasons, but if all else fails then we'll probably do that.

**Comment by axw** [comment]
I've put up a PR which reintroduces the apm-server user: https://github.com/elastic/apm-server/pull/9496. This should be fixed in 8.5.1.
